core/src/main/scala/org/apache/spark/launcher/LauncherBackend.scala

```diff
@@ -71,6 +100,9 @@ private[spark] abstract class LauncherBackend {
     if (connection != null && lastState != state) {
       connection.send(new SetState(state))
       lastState = state
+      if (!_isConnected && stopOnShutdown) {
```

Comment:
The `LauncherBackend` is supposed to run within `Spark App` JVM.  The `connection` is `LauncherConnection` used to report back Spark Job status to User App.  So, if this connection is lost - the User App requesting/waiting for the job completion is no longer available.  So User app is not available shutdown flag would decide whether to kill the Spark job or not.